nestle.py

```diff
@@ -920,13 +1018,18 @@ def prior_transform(u):
         pointvol = expected_vol / npoints
 
         # Update the sampler based on the current active points.
-        if since_update >= update_interval:
+        # For parallel processing, also finishes evaluating
+        # all ongoing trial points to ensure no likelihood call
+        # is wasted.
+        should_update = since_update >= update_interval
+        if should_update and not sampler.queue:
             sampler.update(pointvol)
             since_update = 0
+            should_update = False
 
         # Choose a new point from within the likelihood constraint
         # (having logl > loglstar).
-        u, v, logl, nc = sampler.new_point(loglstar)
+        u, v, logl, nc = sampler.new_point(loglstar, should_update)
```

Comment:
Yes, this is a concern. This problem doesn't end up occurring because of the bug you identified above but will be a concern if I fix things.